paasta_tools/cli/cmds/status.py

```diff
@@ -722,6 +730,66 @@ def build_smartstack_backends_table(backends):
     return format_table(rows)
 
 
+def get_envoy_status_human(
+    registration, expected_backends_per_location, locations
+) -> List[str]:
+    if len(locations) == 0:
+        return [f"Envoy: ERROR - {registration} is NOT in Envoy at all!"]
+
+    output = ["Envoy:"]
+    output.append(f"  Service Name: {registration}")
+    output.append(f"  Backends:")
+    for location in locations:
+        backend_status = envoy_backend_report(
+            expected_backends_per_location, location.running_backends_count
+        )
+        output.append(f"    {location.name} - {backend_status}")
+
+        if location.backends:
+            color = (
+                PaastaColors.green
+                if location.is_proxied_through_casper
+                else PaastaColors.grey
+            )
+            is_proxied_through_casper_output = color(
+                f"{location.is_proxied_through_casper}"
+            )
+            output.append(
+                f"      Proxied through Casper: {is_proxied_through_casper_output}"
+            )
+
+            backends_table = build_envoy_backends_table(location.backends)
+            output.extend([f"      {line}" for line in backends_table])
+
+    return output
+
+
+def build_envoy_backends_table(backends):
```

Comment:
Here and below, for all added/updated functions, could you please add typing annotation for the arg and the return value.